src/ast2ram/seminaive/ClauseTranslator.cpp

```diff
@@ -269,38 +269,41 @@ Own<ram::Operation> ClauseTranslator::addAdtUnpack(
         Own<ram::Operation> op, const ast::BranchInit* adt, int curLevel) const {
     assert(!context.isADTEnum(adt) && "ADT enums should not be unpacked");
 
-    // set branch tag constraint
-    op = addEqualityCheck(std::move(op), mk<ram::TupleElement>(curLevel, 0),
-            mk<ram::SignedConstant>(context.getADTBranchId(adt)), false);
-
     std::vector<ast::Argument*> branchArguments;
-    auto dummyArg = mk<ast::UnnamedVariable>();
 
+    int branchLevel;
     if (context.isADTBranchSimple(adt)) {
+        // for ADT with arity < 2, we have a single level
+        branchLevel = curLevel;
         // only for ADT with arity less than two (= simple)
         // add padding for branch id
+        auto dummyArg = mk<ast::UnnamedVariable>();
```

Comment:
The dummyArg will be destroyed after leaving this scope; it must be declared outside this scope.